docs/security.rst

```diff
@@ -5,6 +5,39 @@ We take the security of ``cryptography`` seriously. The following are a set of
 policies we have adopted to ensure that security issues are addressed in a
 timely fashion.
 
+What is a security issue?
+-------------------------
+
+Anytime it's possible to write code using ``cryptography``'s public API which
+does not provide the guarantees that a reasonable developer would expect it to
+based on our documentation.
+
+That's a bit academic, but basically it means the scope of what we consider a
+vulnerability is broad, and we do not require a proof of concept or even a
+specific exploit, merely a reasonable threat model under which ``cryptography``
+could be attacked.
+
+To give a few examples of things we would consider security issues:
+
+* If a recipe, such as Fernet, made it easy for a user to bypass
+  confidentiality or integrity with the public API (e.g. if the API let a user
+  reuse nonces).
+* If, under any circumstances, we used a CSPRNG which wasn't fork-safe.
+* If ``cryptography`` used an API in an underlying C library and failed to
+  handle error conditions safely.
+
+Examples of things we wouldn't consider security issues:
+
+* Offering ECB mode for symmetric encryption. Though ECB is critically weak, it
+  is documented as being weak in our documentation.
```

Comment:
Good point.

On Fri, Feb 19, 2016 at 7:09 AM, Donald Stufft <notifications@github.com>
wrote:

> In docs/security.rst
> <https://github.com/pyca/cryptography/pull/2725#discussion_r53453458>:
>
> > +specific exploit, merely a reasonable threat model under which ``cryptography``
> > +could be attacked.
> > +
> > +To give a few examples of things we would consider security issues:
> > +
> > +* If a recipe, such as Fernet, made it easy for a user to bypass
> > +  confidentiality or integrity with the public API (e.g. if the API let a user
> > +  reuse nonces).
> > +* If, under any circumstances, we used a CSPRNG which wasn't fork-safe.
> > +* If ``cryptography`` used an API in an underlying C library and failed to
> > +  handle error conditions safely.
> > +
> > +Examples of things we wouldn't consider security issues:
> > +
> > +* Offering ECB mode for symmetric encryption. Though ECB is critically weak, it
> > +  is documented as being weak in our documentation.
>
> Should this explicitly mention offering ECB *in the hazmat layer*? I
> believe we would treat it as a security issue if the recipes layer allowed
> you to use ECB somehow yea?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2725/files#r53453458>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084